File: JInstantSearch/app/src/main/java/com/woowrale/architecture/mvp/ui/search/home/SearchPresenter.java
package com.woowrale.architecture.mvp.ui.search.home;

import com.woowrale.architecture.mvp.ui.base.BasePresenter;

public class SearchPresenter extends BasePresenter<SearchView> {

    private SearchActivity searchActivity;

    public SearchPresenter(SearchActivity searchActivity) {
        this.searchActivity = searchActivity;
    }

    public void navigateToSearch(String searchType){
        if (searchType.equals("L")){
            searchActivity.navigateToSearchLocal();
        }else{
            searchActivity.navigateToSearchRemote();
        }
    }
}

File: AndroidXPatternsApp/kmvp-app/src/main/java/com/woowrale/kmvp/ui/search/home/SearchPresenter.kt
package com.woowrale.kmvp.ui.search.home

import com.woowrale.kmvp.ui.base.BasePresenter
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class SearchPresenter @Inject constructor() : BasePresenter<SearchView>() {

    fun navigateToSearch(searchType: String) {
        if (searchType == "L") {
            getView().navigateToSearchLocal()
        } else {
            getView().navigateToSearchRemote()
        }
    }
}
